src/media/js/elements/select.js

```diff
@@ -0,0 +1,263 @@
+                        if (this.alignOptions() > 0) {
+                            this.setAttribute('data-mkt-select--aligned', true);
+                        }
+                    }
+                    if (bool) {
+                        this.classList.toggle(VISIBLE, bool);
+                    } else {
+                        this.classList.toggle(VISIBLE);
+                    }
+                    return this;
+                },
+            },
+            value: {
+                get: function() {
+                    return this.select.value;
+                },
+                set: function(value) {
+                    // Change value of hidden select element.
+                    var root = this;
+                    var select = root.select;
+                    select.value = value;
+
+                    // Change [selected] of <option>s.
+                    var options = select.querySelectorAll('option');
+                    var selectedOption = select.querySelector(
+                        'option[value="' + value +'"]');
+                    for (var i = 0; i < options.length; i++) {
+                        options[i].removeAttribute('selected');
+                    }
+                    selectedOption.setAttribute('selected', '');
+
+                    // Change text.
+                    root.querySelector('mkt-selected-text').innerHTML =
+                        selectedOption.innerHTML;
+
+                    // Change [selected] of <mkt-option>s.
+                    var mktOptions = root.querySelectorAll('mkt-option');
+                    for (i = 0; i < mktOptions.length; i++) {
+                        mktOptions[i].update();
+                    }
+
+                    // Update [data-mkt-option--index]es.
+                    this.updateOptionIndices();
+                    $(root).trigger('change');
+                }
+            },
+            alignOptions: {
+                value: function() {
+                    // Aligns options with the selected/visible top option.
+                    var offset = $('mkt-selected-text', this).position().left;
+                    $('mkt-option', this).css('padding-left', offset);
+                    return offset;
+                }
+            },
+            updateOptionIndices: {
+                value: function() {
+                    // Sets a data-index on each mkt-option that is not
+                    // [selected]. Prominently used to set colors since we
+                    // don't have CSS4 // nth-match.
+                    var selectedOption = this.querySelector(
+                        'mkt-option[selected]');
+                    var mktOptions = this.querySelectorAll('mkt-option');
+                    var index = 0;
+                    for (var i = 0; i < mktOptions.length; i++) {
+                        var mktOption = mktOptions[i];
+                        if (mktOption == selectedOption) {
```

Comment:
I feel like it makes more sense to check `mktOption.hasAttribute('selected')` instead of doing an object comparison from the extra query.